src/main/java/seedu/duke/ExpenseTracker.java

```diff
@@ -0,0 +1,320 @@
+        this.lastResetMonth = -1;
+    }
+
+    // For testing purposes
+    public List<Expense> getExpenses() {
+        return expenses;
+    }
+
+    public List<Category> getCategories() {
+        return categories;
+    }
+
+    public Map<Category, Budget> getBudgets() {
+        return budgets;
+    }
+
+    public void toggleAutoReset() {
+        isautoResetEnabled = !isautoResetEnabled;
+        System.out.println("Automatic budget reset is now " + (isautoResetEnabled ? "ON" : "OFF") + ".");
+    }
+
+    /**
+     * Resets the budget limits for all categories.
+     *
+     * This method iterates over all budgets in the tracker and resets each budget's
+     * limit as per the current configuration. By default, it maintains the same limit
+     * for each budget, but the reset logic can be adjusted if needed.
+     */
+    private void resetBudgets() {
+        for (Budget budget : budgets.values()) {
+            // Resetting the budget logic can be adjusted as needed
+            budget.setLimit(budget.getLimit()); // For now, just maintaining the same limit
+        }
+        System.out.println("Budgets have been reset for all categories.");
+    }
+
+    /**
+     * Checks if it is a new month and resets budgets if auto-reset is enabled.
+     *
+     * This method uses the current month to determine if a monthly budget reset
+     * should occur. If auto-reset is enabled and the month has changed since the
+     * last reset, it triggers a reset of all budgets and updates the last reset month.
+     */
+    public void checkAndResetBudgets() {
+        Calendar calendar = Calendar.getInstance();
+        int currentMonth = calendar.get(Calendar.MONTH);
+
+        if (isautoResetEnabled && currentMonth != lastResetMonth) {
+            resetBudgets();
+            lastResetMonth = currentMonth;  // Update last reset month
+        }
+    }
+
+    /**
+     * Manages the monthly budget reset process.
+     *
+     * This method is intended to be called periodically to ensure budgets are reset
+     * at the start of a new month if necessary. It delegates the actual reset logic
+     * to the checkAndResetBudgets method, which handles auto-reset checks.
+     */
+    public void manageMonthlyReset() {
+        checkAndResetBudgets();
+    }
+
+
+    private String formatInput(String input) {
+        if (input == null || input.isEmpty()) {
+            return input;
+        }
+        return input.substring(0, 1).toUpperCase() + input.substring(1).toLowerCase();
+    }
+
+    public void addExpense(String name, double amount, String categoryName) {
+        String formattedCategoryName = formatInput(categoryName.trim());
+        Category existingCategory = null;
+        for (Category category : categories) {
+            if (category.getName().equalsIgnoreCase(formattedCategoryName)) {
+                existingCategory = category;
+                break;
+            }
+        }
+        if (existingCategory == null) {
```

Comment:
The logic to find an existing category is repeated in multiple methods (addExpense, setBudgetLimit). Consider refactoring this into a private helper method to reduce redundancy.